src/map/zone.cpp

```diff
@@ -1081,8 +1081,8 @@ void CZone::CharZoneIn(CCharEntity* PChar)
         auto* PBattlefield = m_BattlefieldHandler->GetBattlefield(PChar, true);
         if (PBattlefield != nullptr && PChar->StatusEffectContainer->HasStatusEffectByFlag(EFFECTFLAG_CONFRONTATION))
         {
-            bool isEntered = PChar->StatusEffectContainer->GetStatusEffect(EFFECT_BATTLEFIELD)->GetSubPower() == 1;
-            PBattlefield->InsertEntity(PChar, isEntered);
+            bool isRegistered = PBattlefield->IsRegistered(PChar);
```

Comment:
The battlefield status effect subpower is used to track if the player is actively in the battlefield or sitting outside of it.
Imagine the scenario where a player goes into a battlefield and a party member is outside but before entering disconnects or purposefully logs out real quick. They are suppose to be able to reconnect and still be able to enter the battlefield. That is exactly what the `isEntered` is trying to handle. By using `isRegistered` instead it will add a player into the battlefield even if they haven't entered it yet. This will undoubtedly cause some weird behaviors.

Reply:
Ah, thanks for the context! That's really helpful information. I updated the pull request with less code; the WARPDC leavecode should handle warp/teleport cases in the code in CZone::CharZoneIn where it checks if the player has landed in a non-Battlefield zone, so I think this is okay? I did notice though that with this code added the leavecode was always landing as WARPDC even when just walking out of the battlefield.

I reset my server to be with staging and I can walk out and back into battlefields as long as there's a player in there, and I can log out, have the other player zone away, and zone back in without getting kicked out. I saw a PR from you a while ago that I feel like addressed both of those cases specifically, do you think something went wrong somewhere? I checked against LSB and the code seems pretty similar, only things missing in ASB seemed like improvements, but I don't think I saw anything that really pointed me in the direction of a break.

Reply:
I thought this was all working so I was initially thinking it was a regression due to some other changes I made but now it sounds like it might be an ASB merge issue? I can test it out to see if it occurs on LSB. Does the disconnected character need to be inside or outside the battlefield when disconnecting?

Reply:
I’ve only tried disconnecting from within an instance, which is what happened during internal testing yesterday

Reply:
Latest version works for the getting booted after reconnecting to a stale battlefield, reconnecting during an active battlefield and being able to participate, and preventing players from exiting the battlefield and re-entering.

However, I cannot enter a battlefield if one character disconnects while outside for either this branch or staging; the character teleports inside to a blank battlefield and gets the following error:
![image](https://user-images.githubusercontent.com/8849608/205475224-81060f09-874c-48e5-83e8-5688e7bb514d.png)
Tested on Kindred Spirits in Throne Room.